tests/experiment/test_procedure.py

```diff
@@ -88,8 +93,34 @@ class TestProcedure(Procedure):
         @property
         def prop(self):
             return self.x
+
     p = TestProcedure()
     with pytest.raises(AttributeError):
         _ = p.prop  # AttributeError
     p.x = 5
     assert p.prop == 5
+
+
+@pytest.mark.parametrize("header, units", (
+        ("x (m)", ureg.m),
+        ("x (m/s)", ureg.m / ureg.s),
+        ("x (V/(m*s))", ureg.V / ureg.m / ureg.s),
+        ("x (1)", ureg.dimensionless)
+))
+def test_procedure_parse_columns(header, units):
+    assert Procedure.parse_columns([header])[header] == ureg.Quantity(1, units)
+
+
+@pytest.mark.parametrize("valid_header_no_unit", (
+        ["x"], ["x ( x + y )"], ["x ( notes )"], ["x [V]"]
+))
+def test_procedure_no_parsed_units(valid_header_no_unit):
+    assert Procedure.parse_columns(valid_header_no_unit) == {}
+
+
+@pytest.mark.parametrize("invalid_header_unit", (
+        ["x (sqrt)"], ["x (x)"], ["x (y)"],
+))
+def test_procedure_invalid_parsed_unit(invalid_header_unit):
+    with pytest.raises(ValueError):
+        Procedure.parse_columns(invalid_header_unit)
```

Comment:
The fact that `"x ( notes )"` pass (without parsed units) and `"x (sqrt)"` fail/raise could be useful for your researcher wanting to put non-unit content into parentheses?

Reply:
Yup exactly, that's why I put them in as tests to call out how notes or equations can be put into a DATA_COLUMNS header.